pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2150,6 +2152,14 @@ public void updateLogicalTableConfig(LogicalTableConfig logicalTableConfig)
       updateBrokerResourceForLogicalTable(logicalTableConfig, tableName);
     }
 
+    @Nullable TimeBoundaryConfig oldTimeBoundaryConfig = oldLogicalTableConfig.getTimeBoundaryConfig();
```

Comment:
(nit) We don't usually annotate local variable. IDE should be able to read the annotation from the method signature